Include Golioth Zephyr SDK docs as a submodule

szczys opened the pull request:
@ChrisGammell will you please test this out and let me know if it works for you. I believe the following will fetch the files you need:

1. Check out this branch
2. The Goiloth Docs are included as a submodule so you have to pull the latest:
    ```git submodule update --recursive --remote```
3. Test locally:
    ```npm start```

Changes are on the Zephyr install instructions page: http://localhost:3000/docs/zephyr-intro/toolchain-install

Commit:
Added Golioth Docs as a submodule

Commit:
Add Golioth Zephyr SDK install instructions using symbolic links to submodule files

ChrisGammell commented:
Trying the `git submodule update --recursive --remote` command and not seeing anything update on my end. Throws an error when I try to start the docs engine, so I'm guessing it's not pulling in the submodule properly?

Commit:
Ignore .vscode

Commit:
Import golioth registration pages from golioth docs submodule

Commit:
Update submodule hash and add correct commands to readme

szczys commented:
Sorry about that, I had the wrong commands there. I've corrected them in the readme.

I also added the Golioth registration/device creation process as imports of the docs submodule pages. I did this a bit differently, storing the files in a `_partials` folder in that directory because I think it makes more sense.

For the Zephyr install tools, I thought about doing the same but didn't because preserving the original path means we can easily copy/paste updates to the setup page that calls those partials.
